sim/src/main/scala/spinal/sim/XSimBackend.scala

```diff
@@ -84,18 +84,31 @@ class XSimBackend(config: XSimBackendConfig) extends Backend {
       ".sh"
     }
   }
-  val simulatePath = new File(s"${workPath}/${projectName}.sim/sim_1/behav/xsim").getAbsolutePath
-  val simulateKernelPath = new File(s"${simulatePath}/xsim.dir/${toplevelName}_behav").getAbsolutePath
-  val compilePath = s"${simulatePath}/compile.${scriptSuffix}"
-  val elaboratePath = s"${simulatePath}/elaborate.${scriptSuffix}"
-
-  val vivadoInstallPath = sys.env.get("VIVADO_HOME") match {
-    case Some(x) => x
-    case None => {
-      assert(assertion = false, "VIVADO_HOME not found! Setup your vivado environment first.")
+  val simulateDir        = s"${workPath}/${toplevelName}_xsim.sim/sim_1/behav/xsim"
+  val simulatePath       = new File(simulateDir).getAbsolutePath
+  val compilePath        = s"${simulatePath}/compile.${scriptSuffix}"
+  val elaboratePath      = s"${simulatePath}/elaborate.${scriptSuffix}"
+
+  val msys2Shell = {
+    if (isWindows) {
+      val msys2ShellPath =
+        sys.env.getOrElse("MSYS2_ROOT", "C:\\msys64") + "\\msys2_shell.cmd"
+      if (!new File(msys2ShellPath).exists) {
```

Comment:
this part is what I concerned.

Reply:
indeed I completely missed that this would fail on msys2 shell, thanks for spotting it!

`msys2Shell` would get :
- `sh` if sbt run by msys (check with `sys.env.get("MSYSTEM")` ?)
- the full `%MSYS2_ROOT%\\msys2_shell.cmd -defterm -here -no-start -mingw64` if ran by cmd

would that be good enough ?

pretty unsure about checking msys vs cmd with only `MSYSTEM` envvar, 
thought about checking if `sh` exists, but there can be some other `sh` in cmd PATH